pyproject.toml

```diff
@@ -24,22 +24,28 @@ tenacity = "^8.0.1"
 cryptography = "^38.0.0"
 jsonschema = "^4.15.0"
 
-[tool.poetry.dev-dependencies]
-coverage = {extras = ["toml"], version = "^6.3.2"}
-flake8 = "^4.0.1"
-pytest = "^7.1.1"
+[tool.poetry.group.fmt.dependencies]
+black = "^22.10.0"
 isort = "^5.10.1"
-black = "^22.3.0"
-pyproject-flake8 = "^0.0.1-alpha.3"
+
+[tool.poetry.group.lint.dependencies]
+flake8 = "^5.0.4"
 flake8-docstrings = "^1.6.0"
-flake8-copyright = "^0.2.2"
-flake8-builtins = "^1.5.3"
-pep8-naming = "^0.12.1"
-codespell = "^2.1.0"
-juju = "^2.9.8"
+flake8-copyright = "^0.2.3"
+flake8-builtins = "^2.0.1"
+pyproject-flake8 = "^5.0.4.post1"
+pep8-naming = "^0.13.2"
+codespell = "^2.2.2"
+
+[tool.poetry.group.unit.dependencies]
+coverage = {extras = ["toml"], version = "^6.5.0"}
+pytest = "^7.1.1"
+
+[tool.poetry.group.integration.dependencies]
+juju = "^2"
```

Comment:
Are we using semver 2 for a reason? Safety against changes from 3.0 bugs?

Reply:
Looks like charms built on libjuju 3 cannot work on juju 2 - I see this issue https://bugs.launchpad.net/juju/+bug/1995131
Unfortunately, we can't move to juju 3 because of scale down issue - https://bugs.launchpad.net/juju/+bug/1977582

so I hardcoded juju 2.x

Reply:
Yup, thought so, thanks for confirming.